### PR TITLE
Add kube-version annotations to istio 1.10

### PR DESCRIPTION
**Pull request by doflamingo721**
Linked issue:
https://github.com/rancher/rancher/issues/37697

### CODE REVIEW

File: release.yaml
Comment (aiyengar2) [inline release.yaml:19]: release.yaml should be deleted and recreated only containing these changes since it's an OOB release; all the other charts should be removed.

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
Need to remove other entries from the release.yaml for a new OOB release, otherwise LGTM

**Comment by aiyengar2** [comment]
Assigning @PennyScissors as a second reviewer since this is an OOB release

File: charts/rancher-istio/100.0.0+up1.10.4/Chart.yaml
Comment (aiyengar2) [inline charts/rancher-istio/100.0.0+up1.10.4/Chart.yaml:5]: one more thing; this currently indicates we should not be able to install it on a 1.22 or less cluster, but the issue seems to indicate we only need to drop support for 1.23+? https://github.com/rancher/rancher/issues/37697

Also, why are we limiting to 1.18+ / where does that requirement come from?

File: release.yaml
Comment (doflamingo721) [reply release.yaml:1]: Updated the release.yaml

**Review by aiyengar2 — APPROVED** [review]
LGTM! Waiting on @PennyScissors to approve before merge

**Review by pennyscissors — APPROVED** [review]

**Pull request merged**
